plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts

```diff
@@ -131,7 +129,5 @@ export const useCollection = < T, IsCollection extends boolean = false >(
 		currentResults.current = results;
 	}
 
-	return currentResults.current as IsCollection extends true
-		? { results: T[]; isLoading: boolean }
-		: { results: T; isLoading: boolean };
+	return currentResults.current as { results: T; isLoading: boolean };
```

Comment:
We don't need typecasting here anymore. Also, we will need to restore the comment of this hook. I suggest reverting the whole file to this point: https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts for simplicity.

Reply:
We also need to revert the change we made for callers of this hook too. Like one in `useStoreProducts`.

Reply:
> We don't need typecasting here anymore. Also, we will need to restore the comment of this hook. I suggest reverting the whole file to this point:

It's not possible. If we do, we'll again introduce hundreds of TS errors using the useCollectionData() hook.

Whether we like it or not, the current code allows useCollection() to return not only a [collection of items](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection.ts#L55) but also an object. [Here](https://github.com/woocommerce/woocommerce/blob/ebb300c1e7288b869cba259feb6279bc6cc6fc8e/plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts#L158), the useCollectionData gets an object.

The underlying problem is, again, that the useCollection() hook sometimes does not return a collection, but an object.

Reply:
> We also need to revert the change we made for callers of this hook too. Like one in `useStoreProducts`.

We need to do this if we remove the typecasting.
The final solution is to decouple the useCollection() from the userCollectionData(). imo.